examples/node_watcher.rs

```diff
@@ -16,12 +19,12 @@ async fn main() -> anyhow::Result<()> {
     let events: Api<Event> = Api::all(client.clone());
     let nodes: Api<Node> = Api::all(client.clone());
 
-    let mut obs = Observer::new(nodes)
-        .params(ListParams::default().labels("beta.kubernetes.io/os=linux"))
-        .backoff(ExponentialBackoff::default()) // infinite backoff
-        .watch_applies()
-        .boxed();
+    let obs = try_flatten_applied(StreamBackoff::new(
+        watcher(nodes, ListParams::default().labels("beta.kubernetes.io/os=linux")),
+        ExponentialBackoff::default(),
+    ));
```

Comment:
Since we've been having chats about this outside of github, I thought I would highlight a temporary compromise here for now for anyone else reading it.

- This api is not what we want long term
- the flow of the program should be understood by reading left -> right (not like here where we have to start in the middle, and then descend leftwards, while keeping track of arguments and closing parens)
- the individual pieces presented in this pr are still sensible
- we can fix the flow in the future with stream extensions

The long term goal is to arrive at something like `watcher(api, lp).backoff(backoff).watch_applies()`